src/indexApi.ts

```diff
@@ -150,45 +178,54 @@ async function main(): Promise<void> {
     }
   })
 
-  app.get('/v1/checkTx/', async function(req, res) {
+  app.post('/v1/checkTxs/', async function(req, res) {
     let queryResult
     try {
-      queryResult = asCheckTxReq(req.query)
+      queryResult = asCheckTxsReq(req.body)
     } catch (e) {
-      res.status(400).send(`Missing Request fields.`)
-      return
+      return res.status(400).send(`Missing Request fields.`)
     }
-    const searchedAppId = apps.find(app => app._id === req.query.apiKey)
+    const searchedAppId = apps.find(app => app._id === queryResult.apiKey)
     if (typeof searchedAppId === 'undefined') {
-      res.status(404).send('Api Key has no match.')
-      return
+      return res.status(400).send(`API Key has no match.`)
     }
-    const appId = searchedAppId.appId
-    let result
-    try {
-      const query = `${appId}_${queryResult.pluginId}:${queryResult.orderId}`
-      const dbResult = await reportsTransactions.get(query.toLowerCase())
-      result = asDbTx(dbResult)
-    } catch (e) {
-      console.log(e)
-      if (e != null && e.error === 'not_found') {
-        res.status(404).send(`Could not find transaction.`)
-        return
-      } else {
-        res.status(500).send(`Internal Server Error.`)
-        return
+    const { appId } = searchedAppId
+    const data: CheckTxsResponse[] = []
+    let counter = 0
+    while (counter < queryResult.data.length) {
+      const currentBatch = queryResult.data.slice(
+        counter,
+        counter + CHECKTXS_BATCH_LIMIT
```

Comment:
The batching is supposed to be enforced at the API level meaning the utility server should send no more than 100 order IDs to the report server. Within the checkTxs endpoint, you do not need to batch queries to the database. However you should throw a 400 error if the caller sends more than 100 order IDs.